pkg/backup/restore/restore_manager.go

```diff
@@ -66,6 +69,80 @@ func (rm *restoreManager) UpdateCondition(restore *v1alpha1.Restore, condition *
 	return rm.statusUpdater.Update(restore, condition, nil)
 }
 
+func (rm *restoreManager) pitrEnable(tc *v1alpha1.TidbCluster) error {
+	if !tc.Status.TiKV.PiTRStatus.IsInactive() {
+		return nil
+	}
+	s := v1alpha1.ConfigUpdateStrategyInPlace
+	tc.Spec.TiKV.ConfigUpdateStrategy = &s
+	tc.Status.TiKV.PiTRStatus.NoTask = false
+	originalCfg := new(v1alpha1.PiTROverriddenConfig)
+	if tc.Spec.TiKV.Config == nil || tc.Spec.TiKV.Config.MP == nil {
+		tc.Spec.TiKV.Config = v1alpha1.NewTiKVConfig()
+	}
+	val := tc.Spec.TiKV.Config.Get(TiKVConfigGCThreshold)
+	if val != nil {
+		v := val.MustFloat()
+		originalCfg.GCRatioThreshold = &v
+	}
+	tc.Status.TiKV.PiTRStatus.OriginConfigMap = originalCfg
+	tc.Spec.TiKV.Config.Set(TiKVConfigGCThreshold, -1.0)
+	_, err := rm.deps.TiDBClusterControl.Update(tc)
+	return err
+}
+
+func (rm *restoreManager) pitrDisable(tc *v1alpha1.TidbCluster) error {
+	if tc.Status.TiKV.PiTRStatus.IsInactive() {
+		return nil
+	}
+	threshold := tc.Status.TiKV.PiTRStatus.OriginConfigMap.GCRatioThreshold
+	if threshold != nil {
+		tc.Spec.TiKV.Config.Set(TiKVConfigGCThreshold, threshold)
+	} else {
+		tc.Spec.TiKV.Config.Del(TiKVConfigGCThreshold)
+	}
+	tc.Status.TiKV.PiTRStatus = &v1alpha1.PiTRStatus{}
+	_, err := rm.deps.TiDBClusterControl.Update(tc)
+	return err
+}
+
+func (rm *restoreManager) maybePiTRDisable(tc *v1alpha1.TidbCluster, namespace string) error {
+	restores, err := rm.deps.RestoreLister.Restores(namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if pitrTasksAreDone(restores) {
+		klog.V(2).InfoS("All pitr tasks are done, will disable pitr mode", "namespace", namespace, "tidbcluster", tc.Name)
+		return rm.pitrDisable(tc)
+	}
+	return nil
+}
+
+func pitrTasksAreDone(restores []*v1alpha1.Restore) bool {
+	for _, restore := range restores {
+		if !pitrTaskIsDone(restore) {
+			return false
+		}
+	}
+	return true
+}
+
+func pitrTaskIsDone(restore *v1alpha1.Restore) bool {
+	if v1alpha1.IsRestoreComplete(restore) {
+		return true
+	}
+	if idx, cond := v1alpha1.GetRestoreCondition(&restore.Status, v1alpha1.RestoreFailed); idx != -1 {
+		if time.Since(cond.LastTransitionTime.Time) > 4*time.Hour {
```

Comment:
what's the rational here?

Reply:
A failed restore may not block GC forever.